script/tool/lib/src/publish_plugin_command.dart

```diff
@@ -102,60 +119,55 @@ class PublishPluginCommand extends PluginCommand {
     }
     _print('Local repo is ready!');
 
-    await _publish();
+    final Directory packageDir = _checkPackageDir(package);
+    await _publishPlugin(packageDir: packageDir);
+    await _tagRelease(packageDir: packageDir, remote: remote, remoteUrl: remoteUrl, shouldPushTag: shouldPushTag);
+    await _finishSuccesfully();
+  }
+
+  Future<void> _publishPlugin({@required Directory packageDir}) async {
+    await _checkGitStatus(packageDir);
+    await _publish(packageDir);
     _print('Package published!');
+  }
+
+  Future<void> _tagRelease({@required Directory packageDir, @required String remote, @required String remoteUrl, @required bool shouldPushTag}) async {
     if (!argResults[_tagReleaseOption]) {
-      return await _finishSuccesfully();
+      return;
     }
-
     _print('Tagging release...');
-    final String tag = _getTag();
-    await processRunner.runAndExitOnError('git', <String>['tag', tag],
-        workingDir: _packageDir);
+    final String tag = _getTag(packageDir);
```

Comment:
Let's flip this and the line above, so the print can include the tag for ease of debugging issues.